pkg/asset/installconfig/gcp/validation.go

```diff
@@ -532,6 +532,11 @@ func validateNetworks(client API, ic *types.InstallConfig, fieldPath *field.Path
 }
 
 func validateServiceEndpoints(_ API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	// if this is not an internal install, skip the validation
+	if ic.Publish != types.InternalPublishingStrategy {
+		return nil
+	}
```

Comment:
Why skip validation for public clusters? It seems like this would just allow users installing public clusters to input bad endpoints?

Reply:
Looking at [CORS-2389](https://issues.redhat.com/browse/CORS-2389), which is the parent epic for this story, it appears that the service endpoints would be private. I am not sure if that also means that the cluster would be private by extension.

Reply:
I agree, the use case for endpoints is for private clusters. We should have validation (if there isn't already) to prevent users from specifying endpoints with public clusters, in which case this code could never be executed and we are safe to remove it.

I don't think there should ever be a scenario where: if the endpoints exist, we skip the validation.